apps/system/lockscreen/js/lockscreen.js

```diff
@@ -434,10 +434,14 @@
     // bug 992277. We don't need to use SettingsListener because
     // we're only interested in changes to the setting, and don't
     // keep track of its value.
-    navigator.mozSettings.addObserver('lockscreen.lock-immediately',
-      (function(event) {
-      if (event.settingValue === true) {
+    window.SettingsListener.observe('lockscreen.lock-immediately',
```

Comment:
LockScreen registers its settings observers as part of LockScreen.init() which is only ever called once when System starts here:

http://mxr.mozilla.org/gaia/source/apps/system/lockscreen/js/lockscreen_bootstrap.js#27

Do you think this will require deeper verification?